src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs

```diff
@@ -1049,5 +1061,44 @@ public void IList_NonGeneric_RemoveAt_ZeroMultipleTimes(int count)
         }
 
         #endregion
+
+        #region Enumerator.Current
+
+        // Enumerator.Current should fail at end after new elements was added
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void IList_NonGeneric_CurrentAtEnd_AfterAdd(int count)
```

Comment:
This should have some shared code with the generic version rather than just being copied over.